xcm-simulator/src/tests/mod.rs

```diff
@@ -0,0 +1,499 @@
+	);
+	ParaA::execute_with(|| {
+		assert_ok!(ParachainPalletXcm::send_xcm(
+			Here,
+			(Parent, Parachain(2)),
+			Xcm(vec![Transact {
+				origin_kind: OriginKind::SovereignAccount,
+				require_weight_at_most: Weight::from_parts(INITIAL_BALANCE as u64, 1024 * 1024),
+				call: remark.encode().into(),
+			}]),
+		));
+	});
+
+	ParaB::execute_with(|| {
+		use parachain::{RuntimeEvent, System};
+		assert!(System::events().iter().any(|r| matches!(
+			r.event,
+			RuntimeEvent::System(frame_system::Event::Remarked { .. })
+		)));
+	});
+}
+
+#[test]
+fn reserve_transfer() {
+	MockNet::reset();
+
+	let withdraw_amount = 123;
+
+	Relay::execute_with(|| {
+		assert_ok!(RelayChainPalletXcm::limited_reserve_transfer_assets(
+			relay_chain::RuntimeOrigin::signed(ALICE),
+			Box::new(Parachain(1).into()),
+			Box::new(AccountId32 { network: None, id: ALICE.into() }.into()),
+			Box::new((Here, withdraw_amount).into()),
+			0,
+			Unlimited,
+		));
+		assert_eq!(
+			relay_chain::Balances::free_balance(child_account_id(1)),
+			INITIAL_BALANCE + withdraw_amount
+		);
+	});
+
+	ParaA::execute_with(|| {
+		// free execution, full amount received
+		assert_eq!(
+			pallet_balances::Pallet::<parachain::Runtime>::free_balance(&ALICE),
+			INITIAL_BALANCE + withdraw_amount
+		);
+	});
+}
+
+#[test]
+fn remote_locking_and_unlocking() {
+	MockNet::reset();
+
+	let locked_amount = 100;
+
+	ParaB::execute_with(|| {
+		let message = Xcm(vec![LockAsset {
+			asset: (Here, locked_amount).into(),
+			unlocker: Parachain(1).into(),
+		}]);
+		assert_ok!(ParachainPalletXcm::send_xcm(Here, Parent, message.clone()));
+	});
+
+	Relay::execute_with(|| {
+		use pallet_balances::{BalanceLock, Reasons};
+		assert_eq!(
+			relay_chain::Balances::locks(child_account_id(2)),
+			vec![BalanceLock { id: *b"py/xcmlk", amount: locked_amount, reasons: Reasons::All }]
+		);
+	});
+
+	ParaA::execute_with(|| {
+		assert_eq!(
+			parachain::MsgQueue::received_dmp(),
+			vec![Xcm(vec![NoteUnlockable {
+				owner: (Parent, Parachain(2)).into(),
+				asset: (Parent, locked_amount).into()
+			}])]
+		);
+	});
+
+	ParaB::execute_with(|| {
+		// Request unlocking part of the funds on the relay chain
+		let message = Xcm(vec![RequestUnlock {
+			asset: (Parent, locked_amount - 50).into(),
+			locker: Parent.into(),
+		}]);
+		assert_ok!(ParachainPalletXcm::send_xcm(Here, (Parent, Parachain(1)), message));
+	});
+
+	Relay::execute_with(|| {
+		use pallet_balances::{BalanceLock, Reasons};
+		// Lock is reduced
+		assert_eq!(
+			relay_chain::Balances::locks(child_account_id(2)),
+			vec![BalanceLock {
+				id: *b"py/xcmlk",
+				amount: locked_amount - 50,
+				reasons: Reasons::All
+			}]
+		);
+	});
+}
+
+/// Scenario:
+/// A parachain transfers an NFT resident on the relay chain to another parachain account.
+///
+/// Asserts that the parachain accounts are updated as expected.
+#[test]
+fn withdraw_and_deposit_nft() {
```

Comment:
So we finally need to write test related to nfts?

Reply:
nope no use. They are the tests related to the Para* from the polkadot.sdk. Our test will be included in laosish_xcm.rs file

Reply:
Okay, so maybe we may delete them at some point? I'm not sure if we need them

Reply:
If we modify the configuration of the mock parachains and they begin to fail, we simply remove them. They're not really within our scope or serving our purpose, but since they're harmless tests, there's no strong reason to remove them. They might still reveal some changes in behavior.